sdks/python/apache_beam/testing/load_tests/pardo_test.py

```diff
@@ -132,22 +137,38 @@
 @unittest.skipIf(not load_test_enabled, 'Enabled only for phrase triggering.')
 class ParDoTest(LoadTest):
   def setUp(self):
-    self.output = self.pipeline.get_option('output')
-    self.iterations = self.pipeline.get_option('number_of_counter_operations')
+    super(ParDoTest, self).setUp()
+    self._apply_filter()
+    self.iterations = self._get_option_or_default('iterations')
+    self.nb_of_counter = self._get_option_or_default('number_of_counters')
+    self.nb_of_operations = self._get_option_or_default(
+        'number_of_counter_operations')
+
+  def _apply_filter(self):
+    """Prevents metrics from namespace other than specified in pipeline
+    options from being published."""
+    if self.metrics_monitor is not None:
+      self.metrics_monitor.filters = MetricsFilter().with_namespace(
+          self.metrics_namespace)
+
+  def _get_option_or_default(self, opt_name, default=1):
+    option = self.pipeline.get_option(opt_name)
+    return int(option) if option is not None else default
 
   def testParDo(self):
-    class _GetElement(beam.DoFn):
-      from apache_beam.testing.load_tests.load_test_metrics_utils import count_bytes
-
-      @count_bytes
-      def process(self, element, namespace, is_returning):
-        if is_returning:
-          yield element
-
-    if not self.iterations:
-      num_runs = 1
-    else:
-      num_runs = int(self.iterations)
+    class CounterOperation(beam.DoFn):
```

Comment:
I think it would be good to have CountBytes metrics in `load_test_metrics_utils`. WDYT about making this class parametrised also by name and namespace so it can be reused in future?

Reply:
I don't think this class can be reusable. Take a note that we don't count bytes here, we just increment counters by 1 in order to simulate stressful conditions (pipeline with a lot of metric counters)